assets/scripts/palette/__tests__/UndoRedo.test.js

```diff
@@ -1,98 +1,103 @@
-import { isUndoAvailable, isRedoAvailable } from '../../streets/undo_stack'
-
-jest.mock('../../streets/undo_stack')
-jest.mock('../../store/actions/undo', () => ({
-  undo: jest.fn(),
-  redo: jest.fn()
+import { cleanup, fireEvent } from '@testing-library/react'
+import { renderWithReduxAndIntl } from '../../../../test/helpers/render'
+import UndoRedo from '../UndoRedo'
+import { replaceUndoStack } from '../../store/actions/undo'
+
+// `getRemixOnFirstEdit` is a legacy function, so for the purposes of this
+// test suite it always returns `false` (aka, assume user owns the street)
+jest.mock('../../streets/remix', () => ({
+  getRemixOnFirstEdit: () => false
 }))
 
 describe('UndoRedo', () => {
-  it('renders two buttons', () => {
-    const wrapper = mount(<UndoRedo locale={{}} />)
-    expect(wrapper.find('button').length).toEqual(2)
+  afterEach(() => {
+    cleanup()
   })
 
-  it('checks if undo or redo is available when undo position changes', () => {
-    // `<UndoRedo>` is mounted with initial props
-    const wrapper = mount(<UndoRedo undoPosition={0} undoStack={[]} />)
+  it('renders two buttons', () => {
+    const wrapper = renderWithReduxAndIntl(<UndoRedo />)
+    expect(wrapper.asFragment()).toMatchSnapshot()
+  })
 
-    // Send new props to `<UndoRedo>`
-    // It doesn't really matter what we set props to.
-    wrapper.setProps({
-      undoPosition: 1,
-      undoStack: []
+  it('enables undo button when an undo stack is created', () => {
+    const wrapper = renderWithReduxAndIntl(<UndoRedo />, {
+      initialState: {
+        undo: {
+          stack: [],
+          position: 0
+        }
+      }
     })
+    const undoButton = wrapper.getByTitle('Undo')
+    const redoButton = wrapper.getByTitle('Redo')
 
-    expect(isUndoAvailable).toBeCalled()
-    expect(isRedoAvailable).toBeCalled()
+    // Initial button states are disabled
+    expect(undoButton.disabled).toEqual(true)
+    expect(redoButton.disabled).toEqual(true)
 
-    // Clear mocks for future tests
-    isUndoAvailable.mockClear()
-    isRedoAvailable.mockClear()
-  })
+    // Update undo stack state from "elsewhere", and `<UndoRedo>` should
+    // receive new props. This is akin to a user editing the street once
+    wrapper.store.dispatch(replaceUndoStack([{ foo: 'bar' }], 1))
 
-  it('checks if undo or redo is available when undo stack changes', () => {
-    // `<UndoRedo>` is mounted with initial props
-    const wrapper = mount(<UndoRedo undoPosition={0} undoStack={[]} />)
+    // Undo button state should become enabled
+    // Redo button should remain disabled
+    expect(undoButton.disabled).toEqual(false)
+    expect(redoButton.disabled).toEqual(true)
+  })
 
-    // Send new props to `<UndoRedo>`
-    // It doesn't really matter what we set props to.
-    wrapper.setProps({
-      undoPosition: 0,
-      undoStack: [{ foo: 'bar' }]
+  it('enables redo button when undo stack position is lower than the top of the stack', () => {
```

Comment:
Just to be clear for me and maybe the message needs to be rewritten:
The position is the position for the current user in the Undo stack and when the user has done an undo action, the redo action is available, right? 
> stack position is lower than the top of the stack

I would consider that an implementation detail, what we want to say with this test is that the Redo action is available when there is something to be Redone, right?